internal/errors/option.go

```diff
@@ -15,12 +15,63 @@
 //
 package errors
 
-var (
-	// ErrInvalidOption returns invalid option error.
-	ErrInvalidOption = func(name string, val interface{}) error {
-		if val == nil {
-			return Errorf("invalid option. name: %s, val: nil", name)
-		}
-		return Errorf("invalid option. name: %s, val: %#v", name, val)
-	}
+import (
+	"fmt"
+	"strings"
 )
+
+// ErrInvalidOption represent the invalid option error
+type ErrInvalidOption struct {
+	name string
+	val  interface{}
+	errs []error
+}
+
+func NewErrInvalidOption(name string, val interface{}, errs ...error) error {
+	return &ErrInvalidOption{
+		name: name,
+		val:  val,
+		errs: errs,
+	}
+}
+
+func (e *ErrInvalidOption) Error() string {
+	if len(e.errs) == 0 {
+		return fmt.Sprintf("invalid option, name: %s, val: %#v", e.name, e.val)
+	}
+
+	errStrs := make([]string, 0, len(e.errs))
+	for i := 0; i < len(e.errs); i++ {
+		errStrs[i] = e.errs[i].Error()
+	}
+
+	return fmt.Sprintf("invalid option, name: %s, val: %#v, error: %v", e.name, e.val, strings.Join(errStrs, ", "))
+}
+
+/*
+   ErrCriticalOption
+*/
```

Comment:
What's this?

Reply:
to separate the implementation from different errors